woocommerce/woocommerce-android
# Feature/ducks in row

Closes #6665 



This PR merges a feature branch into trunk. The list of tasks included in this branch can be found [here](https://github.com/orgs/woocommerce/projects/68/).

**No need to review all the code, just the last commit - 0f056dc080a706c89fa556473ada69f8e1f437bc.**

More info about the project on: p91TBi-8Up-p2


I encountered some conflicts when I was merging trunk into this branch. The main one was related to the removal of TakePaymentFragment and its replacement with SelectPaymentMethodFragment. This [PR on trunk fixed an issue with navigation](https://github.com/woocommerce/woocommerce-android/pull/6784) and I had to manually move it in the new SelectPaymentMethodFragment in the last commit - 0f056dc080a706c89fa556473ada69f8e1f437bc.



### Testing instructions

Smoke test Order Detail.

Test navigation - back behavior 

1. Go to Order List
2. Click on + FAB
3. Enter an amount
4. Continue
5. Continue
6. Tap on for example "Cash"
7. Confirm that the order will be marked as paid
8. The app navigates you back to the Order ~detail~ list
9. Press back - notice you are navigated to the dashboard not the SimplePayment flow
